e2e-tests/executor.ts

```diff
@@ -0,0 +1,166 @@
+// Copyright 2023 Parity Technologies (UK) Ltd.
+import { ApiPromise, WsProvider } from '@polkadot/api';
+import { Keyring } from '@polkadot/keyring';
+// import { KeyringPair } from '@polkadot/keyring/types';
+import { cryptoWaitReady } from '@polkadot/util-crypto';
+
+import { delay } from '../scripts/util';
+import { constructApiPromise } from '../src';
+import { balanceTracker, IBalance } from './balance';
+import { KUSAMA_ASSET_HUB_WS_URL, ROCOCO_ALICE_WS_URL, TRAPPIST_WS_URL } from './consts';
+import { startProgressBar, startTestLogger, terminateProgressBar, testResultLogger, updateProgressBar } from './logger';
+import { assetTests, foreignAssetsTests, IndividualTest, liquidPoolsTests, localTests, tests } from './tests';
+import { verification } from './verification';
+
+const executor = async (testCase: string) => {
+	let originWsUrl = '';
+	let destWsUrl = '';
+
+	let testData: IndividualTest[] = [];
+
+	await cryptoWaitReady();
+
+	const keyring = new Keyring({ type: 'sr25519' });
+
+	switch (testCase) {
+		case '--foreign-assets':
+			testData = tests.foreignAssets;
+			break;
+		case '--liquidity-assets':
+			testData = tests.liquidPools;
+			break;
+		case '--local':
+			testData = tests.local;
+			break;
+		case '--assets':
+			testData = tests.assets;
+			break;
+	}
+
+	let n: { [K: string]: Function } = {};
+
+	switch (testCase) {
+		case '--foreign-assets':
+			n = foreignAssetsTests;
+			break;
+		case '--liquidity-assets':
+			n = liquidPoolsTests;
+			break;
+		case '--local':
+			n = localTests;
+			break;
+		case '--assets':
+			n = assetTests;
+			break;
+	}
+
+	let counter: number = 0;
+
+	startTestLogger(testCase);
+
+	const progressBar = startProgressBar(testData, testCase);
+
+	const results: [string, string, string, boolean][] = [];
+
+	for (const t of testData) {
```

Comment:
All of the manual arg management can be handled using either `ParseArgs`, or `Yargs`. Both libraries are good for handling cli input.

Reply:
But IMO, no need to switch now, what you did is fine, just pointing it out as another option to manage cli args.

Reply:
Will look into it, thanks for the suggestion